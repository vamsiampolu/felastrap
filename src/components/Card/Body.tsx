import { useFela } from "react-fela";
import type { DivProps, StyleProps } from "../common";
import { overlayBase, overlayLayoutRule } from "./imageStyle";

const padding = {
  paddingTop: "1rem",
  paddingBottom: "1rem",
  paddingLeft: "1rem",
  paddingRight: "1rem",
};

const body = ({ color }: StyleProps<CardBodyProps>) => ({
  ...padding,
  color,
  flexGrow: 1,
  flexShrink: 1,
  flexBasis: "auto",
});

export type CardBodyProps = DivProps & { color?: string; overlay?: boolean };

export const CardBody = ({
  className = "",
  children,
  color = "#000",
  overlay = false,
  ...other
}: CardBodyProps) => {
  const { css } = useFela({ color, overlay });
  const cx = `${css([body, overlayBase, overlayLayoutRule])} ${
    className || ""
  }`.trim();
  return (
    <div data-testid="card-body" className={cx} {...other}>
      {children}
    </div>
  );
};
