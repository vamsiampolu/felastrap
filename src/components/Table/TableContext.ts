import { createContext } from "react";
import type { Variant } from "../common";

export type TableContextValue = {
  bordered: boolean;
  borderless: boolean;
  variant: Variant;
  striped: boolean;
  hover: boolean;
};

export const TableContext = createContext<TableContextValue>({
  bordered: false,
  borderless: false,
  variant: "default",
  striped: false,
  hover: false,
});
