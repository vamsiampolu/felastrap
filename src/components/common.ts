import type * as Css from "csstype";
import type * as Fela from "fela";
import type {
  ComponentPropsWithoutRef,
  ComponentType,
  ElementType,
} from "react";
import type * as ReactFela from "react-fela";

export type ExcludeFrom<T, U> = T extends U ? never : T;
export type Theme = object;

export type Rule = Fela.TRule;
export type CSSObject = Fela.CSSObject;
export type StyleProps<P = object> = ReactFela.StyleProps<Theme, P>;
export type StyleFunction<P = object> = ReactFela.StyleFunction<Theme, P>;

export type Padding = Required<
  Pick<
    Css.Properties,
    "paddingTop" | "paddingLeft" | "paddingRight" | "paddingBottom"
  >
>;
export type BorderRadius = Required<
  Pick<
    Css.Properties,
    | "borderTopLeftRadius"
    | "borderTopRightRadius"
    | "borderBottomRightRadius"
    | "borderBottomLeftRadius"
  >
>;

export type PaddingString = Required<Css.Properties["padding"]>;
export type Color = Css.Property.Color;

export type HeaderTypes = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
export type Size = "small" | "medium" | "large";
export type Variant =
  | "default"
  | "danger"
  | "info"
  | "primary"
  | "secondary"
  | "success"
  | "warning";

export type FontSize = Css.Property.FontSize;
export type ColorMapping = Record<Variant, Color>;
export type PaddingBySize = Record<Size, Padding>;

export type SpanProps = ComponentPropsWithoutRef<"span">;
export type DivProps = ComponentPropsWithoutRef<"div">;
export type ListItemProps = ComponentPropsWithoutRef<"li">;

export type DropdownDirection = "up" | "down" | "end" | "start" | "centered";
export type CaretDirection = "up" | "down" | "end" | "start";
export type NavVariant =
  | "default"
  | "tabs"
  | "pills"
  | "underline"
  | "vertical";

export type CaretProps = DivProps & {
  width: string | number | null;
};

export type DynamicElement<T extends ElementType = "span"> =
  | keyof JSX.IntrinsicElements
  | ComponentType<T>;
export type PargraphProps = ComponentPropsWithoutRef<"p">;
export type ImageProps = ComponentPropsWithoutRef<"img">;
export type TooltipPlacement = "top" | "left" | "bottom" | "right";

export type MediaQuerySize = "xs" | "sm" | "md" | "lg" | "xl" | "xxl";
export type Timer = ReturnType<typeof setTimeout>;
export type OnClick<T extends ElementType> =
  ComponentPropsWithoutRef<T>["onClick"];
export type StyleKey = Exclude<
  keyof HTMLElement["style"],
  "length" | "parentRule"
>;
