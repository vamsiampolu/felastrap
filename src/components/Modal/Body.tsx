import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  position: "relative",
  padding: "15px",
  borderBottom: "1px solid #dee2e6",
});

const scrollableRule = ({ scrollable = false }) =>
  scrollable
    ? {
        overflowY: "auto !important",
        minHeight: "1500px",
      }
    : {};

export const ModalBody = ({
  className = "",
  children,
  scrollable = false,
  ...props
}: DivProps & { scrollable?: boolean }) => {
  const { css } = useFela({ scrollable });
  const cx = `${css([rule, scrollableRule])} bs-modal-body ${className}`.trim();
  return (
    <div data-testid="modal-body" className={cx} {...props}>
      {children}
    </div>
  );
};
