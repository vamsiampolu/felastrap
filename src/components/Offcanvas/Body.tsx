import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  flexGrow: 1,
  padding: "1rem 1rem",
  overflowY: "auto",
});

export const OffcanvasBody = ({
  className = "",
  children,
  ...props
}: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} offcanvas-body ${className}`.trim();
  return (
    <div data-testid="offcanvas-body" className={cx} {...props}>
      {children}
    </div>
  );
};
