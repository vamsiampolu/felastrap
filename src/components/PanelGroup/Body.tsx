import { useFela } from "react-fela";
import { DivProps } from "../common";
import { Content } from "../Collapse/Content";

const flushBody = ({ flush = false }) => {
  return flush ? { borderWidth: 0 } : {};
};

const rule = () => ({
  padding: "1rem 1.25rem",
});

export type BodyProps = DivProps & { flush?: boolean };
export const Body = ({
  className = "",
  flush = false,
  children,
  ...other
}: BodyProps) => {
  const { css } = useFela({ flush });
  const cx = `${css([rule, flushBody])} panel-group-body ${className}`;
  return (
    <div data-testid="panel-group-body" className={cx} {...other}>
      {children}
    </div>
  );
};

export const BodyContainer = ({ className = "", id, ...props }: DivProps) => {
  return (
    <Content id={id}>
      <Body className={className} {...props} />
    </Content>
  );
};
