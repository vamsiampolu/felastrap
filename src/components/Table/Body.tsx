import { IStyle } from "fela";
import { ComponentPropsWithoutRef, useContext } from "react";
import { useFela } from "react-fela";
import { TableContext } from "./TableContext";
import { hoveredBackgroundColorByVariant } from "./variant";
import { Variant } from "../common";

export type TableBodyProps = ComponentPropsWithoutRef<"tbody"> & {
  striped?: boolean;
  hover?: boolean;
  bordered?: boolean;
};

const rule = () => ({ verticalAlign: "inherit" });

const hoverRule = ({ hover = false, variant = "default" }): IStyle =>
  hover
    ? {
        "> tr:hover > *": {
          cursor: "pointer",
          color: "#000",
          backgroundColor: hoveredBackgroundColorByVariant[variant as Variant],
        },
      }
    : {};

const borderedRule = ({ bordered = false }) =>
  bordered
    ? {
        borderWidth: "1px 0",
      }
    : {};

export const TableBody = ({
  className = "",
  children,
  ...props
}: TableBodyProps) => {
  const { hover, bordered, variant = "default" } = useContext(TableContext);
  const { css } = useFela({ hover, bordered, variant });
  const cx = `${css([
    rule,
    hoverRule,
    borderedRule,
  ])} table-body ${className}`.trim();
  return (
    <tbody data-testid="table-body" className={cx} {...props}>
      {children}
    </tbody>
  );
};
