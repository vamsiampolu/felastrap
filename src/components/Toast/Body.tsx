import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  padding: "0.75rem",
  wordWrap: "break-word",
});

export const ToastBody = ({ children, className = "", ...props }: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} toast-body ${className}`.trim();
  return (
    <div data-testid="toast-body" className={cx} {...props}>
      {children}
    </div>
  );
};
