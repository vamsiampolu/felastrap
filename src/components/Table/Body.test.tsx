import { IRenderer } from "fela";
import { TableBody } from "./Body";
import { TableContext, TableContextValue } from "./TableContext";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../test-helpers";
import { render, screen } from "@testing-library/react";
import { Variant } from "../common";

describe("components/Table/Body", () => {
  const value: TableContextValue = {
    variant: "default",
    striped: false,
    bordered: false,
    borderless: false,
    hover: false,
  };

  const variants = [
    "default",
    "danger",
    "info",
    "primary",
    "secondary",
    "success",
    "warning",
  ];

  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a Table Body", async () => {
    const component = (
      <TableContext.Provider value={value}>
        <table>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <TableBody>
            <tr>
              <td>A</td>
              <td>B</td>
            </tr>
          </TableBody>
        </table>
      </TableContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("table-body")).toBeInTheDocument();

    expect(screen.getByTestId("table-body")).toHaveClass("table-body");
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a Table Body with a className", async () => {
    const component = (
      <TableContext.Provider value={value}>
        <table>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <TableBody className="comfortable-body">
            <tr>
              <td>A</td>
              <td>B</td>
            </tr>
          </TableBody>
        </table>
      </TableContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("table-body")).toBeInTheDocument();

    expect(screen.getByTestId("table-body")).toHaveClass("table-body");
    expect(screen.getByTestId("table-body")).toHaveClass("comfortable-body");
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a hover Table Body", async () => {
    const ctxValue: TableContextValue = { ...value, hover: true };
    const component = (
      <TableContext.Provider value={ctxValue}>
        <table>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <TableBody>
            <tr>
              <td>A</td>
              <td>B</td>
            </tr>
          </TableBody>
        </table>
      </TableContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("table-body")).toBeInTheDocument();

    expect(screen.getByTestId("table-body")).toHaveClass("table-body");
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a bordered Table Body", async () => {
    const ctxValue: TableContextValue = { ...value, bordered: true };
    const component = (
      <TableContext.Provider value={ctxValue}>
        <table>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <TableBody>
            <tr>
              <td>A</td>
              <td>B</td>
            </tr>
          </TableBody>
        </table>
      </TableContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("table-body")).toBeInTheDocument();

    expect(screen.getByTestId("table-body")).toHaveClass("table-body");
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it.each(variants)(
    "should render a Table Body of variant %s",
    async (variant) => {
      const ctxValue: TableContextValue = {
        ...value,
        variant: variant as Variant,
      };
      const component = (
        <TableContext.Provider value={ctxValue}>
          <table>
            <thead>
              <tr>
                <td>Column A</td>
                <td>Column B</td>
              </tr>
            </thead>
            <TableBody>
              <tr>
                <td>A</td>
                <td>B</td>
              </tr>
            </TableBody>
          </table>
        </TableContext.Provider>
      );

      render(<Wrapper renderer={renderer}>{component}</Wrapper>);
      expect(screen.getByTestId("table-body")).toBeInTheDocument();

      expect(screen.getByTestId("table-body")).toHaveClass("table-body");
      const snapshot = await createSnapshot({ component, renderer });
      expect(snapshot).toMatchSnapshot();
    },
  );

  it.each(variants)(
    "should render a hovered Table Body of variant %s",
    async (variant) => {
      const ctxValue: TableContextValue = {
        ...value,
        variant: variant as Variant,
        hover: true,
      };

      const component = (
        <TableContext.Provider value={ctxValue}>
          <table>
            <thead>
              <tr>
                <td>Column A</td>
                <td>Column B</td>
              </tr>
            </thead>
            <TableBody>
              <tr>
                <td>A</td>
                <td>B</td>
              </tr>
            </TableBody>
          </table>
        </TableContext.Provider>
      );

      render(<Wrapper renderer={renderer}>{component}</Wrapper>);
      expect(screen.getByTestId("table-body")).toBeInTheDocument();

      expect(screen.getByTestId("table-body")).toHaveClass("table-body");
      const snapshot = await createSnapshot({ component, renderer });
      expect(snapshot).toMatchSnapshot();
    },
  );

  it.each(variants)(
    "should render a hover bordered Table Body of variant %s",
    async (variant) => {
      const ctxValue: TableContextValue = {
        ...value,
        variant: variant as Variant,
        hover: true,
        bordered: true,
      };

      const component = (
        <TableContext.Provider value={ctxValue}>
          <table>
            <thead>
              <tr>
                <td>Column A</td>
                <td>Column B</td>
              </tr>
            </thead>
            <TableBody>
              <tr>
                <td>A</td>
                <td>B</td>
              </tr>
            </TableBody>
          </table>
        </TableContext.Provider>
      );

      render(<Wrapper renderer={renderer}>{component}</Wrapper>);
      expect(screen.getByTestId("table-body")).toBeInTheDocument();

      expect(screen.getByTestId("table-body")).toHaveClass("table-body");
      const snapshot = await createSnapshot({ component, renderer });
      expect(snapshot).toMatchSnapshot();
    },
  );
});
