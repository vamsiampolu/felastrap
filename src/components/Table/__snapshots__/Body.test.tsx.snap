// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Table/Body should render a Table Body 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a Table Body of variant danger 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a Table Body of variant default 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a Table Body of variant info 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a Table Body of variant primary 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a Table Body of variant secondary 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a Table Body of variant success 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a Table Body of variant warning 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a Table Body with a className 1`] = `
".a {
  vertical-align: inherit;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a table-body comfortable-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a bordered Table Body 1`] = `
".a {
  vertical-align: inherit;
}
.b {
  border-top-width: 1px;
}
.c {
  border-right-width: 0;
}
.d {
  border-bottom-width: 1px;
}
.e {
  border-left-width: 0;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d e table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hover Table Body 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(0, 0, 0, 0.075);
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hover bordered Table Body of variant danger 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(229, 199, 202, 0.075);
}
.e {
  border-top-width: 1px;
}
.f {
  border-right-width: 0;
}
.g {
  border-bottom-width: 1px;
}
.h {
  border-left-width: 0;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d e f g h table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hover bordered Table Body of variant default 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(0, 0, 0, 0.075);
}
.e {
  border-top-width: 1px;
}
.f {
  border-right-width: 0;
}
.g {
  border-bottom-width: 1px;
}
.h {
  border-left-width: 0;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d e f g h table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hover bordered Table Body of variant info 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(191, 226, 233, 0.075);
}
.e {
  border-top-width: 1px;
}
.f {
  border-right-width: 0;
}
.g {
  border-bottom-width: 1px;
}
.h {
  border-left-width: 0;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d e f g h table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hover bordered Table Body of variant primary 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(191, 209, 236, 0.075);
}
.e {
  border-top-width: 1px;
}
.f {
  border-right-width: 0;
}
.g {
  border-bottom-width: 1px;
}
.h {
  border-left-width: 0;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d e f g h table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hover bordered Table Body of variant secondary 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(209, 210, 212, 0.075);
}
.e {
  border-top-width: 1px;
}
.f {
  border-right-width: 0;
}
.g {
  border-bottom-width: 1px;
}
.h {
  border-left-width: 0;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d e f g h table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hover bordered Table Body of variant success 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(193, 214, 204, 0.075);
}
.e {
  border-top-width: 1px;
}
.f {
  border-right-width: 0;
}
.g {
  border-bottom-width: 1px;
}
.h {
  border-left-width: 0;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d e f g h table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hover bordered Table Body of variant warning 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(236, 225, 190, 0.075);
}
.e {
  border-top-width: 1px;
}
.f {
  border-right-width: 0;
}
.g {
  border-bottom-width: 1px;
}
.h {
  border-left-width: 0;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d e f g h table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hovered Table Body of variant danger 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(229, 199, 202, 0.075);
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hovered Table Body of variant default 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(0, 0, 0, 0.075);
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hovered Table Body of variant info 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(191, 226, 233, 0.075);
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hovered Table Body of variant primary 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(191, 209, 236, 0.075);
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hovered Table Body of variant secondary 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(209, 210, 212, 0.075);
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hovered Table Body of variant success 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(193, 214, 204, 0.075);
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Body should render a hovered Table Body of variant warning 1`] = `
".a {
  vertical-align: inherit;
}
.b > tr:hover > * {
  cursor: pointer;
}
.c > tr:hover > * {
  color: #000;
}
.d > tr:hover > * {
  background-color: rgba(236, 225, 190, 0.075);
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody data-testid="table-body" class="a b c d table-body">
    <tr>
      <td>A</td>
      <td>B</td>
    </tr>
  </tbody>
</table>
;
"
`;
